internal/schema/schema_test.go

```diff
@@ -165,6 +165,40 @@ func TestParse(t *testing.T) {
 				return nil
 			},
 		},
+		{
+			name: "Type extension works correctly",
+			sdl: `
+			type Query {
+				hello: String!
+			}
+
+			extend type Query {
+				world: String!
+			}`,
+			validateSchema: func(s *schema.Schema) error {
+				typ, ok := s.Types["Query"].(*schema.Object)
+				if !ok {
+					return fmt.Errorf("type %q not found", "Query")
+				}
+
+				helloField := typ.Fields.Get("hello")
+				if helloField == nil {
+					return fmt.Errorf("field %q not found", "hello")
+				}
+				if helloField.Type.String() != "String!" {
+					return fmt.Errorf("field %q has an invalid type: %q", "hello", helloField.Type.String())
+				}
+
+				worldField := typ.Fields.Get("world")
+				if worldField == nil {
+					return fmt.Errorf("field %q not found", "world")
+				}
+				if worldField.Type.String() != "String!" {
+					return fmt.Errorf("field %q has an invalid type: %q", "world", worldField.Type.String())
+				}
+				return nil
+			},
+		},
```

Comment:
I want to see some more unit tests with more extension examples before merging this.

Reply:
I'll merge this and I'll add the missing unit tests myself. Thank you @pzduniak